airflow/ui/src/layouts/Details/Grid/Grid.tsx

```diff
@@ -70,9 +72,41 @@ export const Grid = () => {
     [structure?.nodes, openGroupIds],
   );
 
+  // return all groupIds with children in the tree for expanding/collapsing functionality
+  const allGroupIds = useMemo(() => {
+    const traverse = (nodes: Array<NodeResponse>): Array<string> =>
+      nodes.flatMap((node) => (node.children ? [node.id, ...traverse(node.children)] : []));
+
+    return traverse(structure?.nodes ?? []);
+  }, [structure?.nodes]);
+
   return (
     <Flex justifyContent="flex-end" mr={3} position="relative" pt={50} width="100%">
       <Box position="absolute" top="150px" width="100%">
+        {allGroupIds.length > 0 && (
```

Comment:
Let's move all of this to it's own `ToggleGroups.tsx` component. Because we will also want to show it on the graph view.